### ISSUE TITLE
Validation for description-like comments?

### ISSUE THREAD

**Issue by adi-zz** [gajus/eslint-plugin-jsdoc#293]
I am able to validate the description of a `function` using [`match-description`](https://github.com/gajus/eslint-plugin-jsdoc#match-description).

```
/** This function description is checked by eslint-plugin-jsdoc. */
function id(i) { return i }

```
But is there a way to validate other (sort of) descriptions using the same reg exp I declared in `match-description`?
```
/** Can eslint-plugin-jsdoc check this class description ? */
class MyClass {
  /** Can eslint-plugin-jsdoc check this field description? */
  myClassField = 1
}

const myObject = {
  /** Can eslint-plugin-jsdoc check this property description? */
  myProp: true
}

/** Can eslint-plugin-jsdoc check this interface description? */
interface MyTypescriptInterface {
}
```

**Comment by brettz9** [comment]
I've submitted #294 which would allow you to add an option to this rule, `{contexts: ['ClassDeclaration']}` and get linting for classes.

From my testing, ESLint does not appear to allow the context to be `Property` or `VariableDeclaration`; it seems to only handle the likes of functions and class declarations, so we cannot apparently support it for property descriptions at this time.

As far as TypeScript, when I tried the following (using the types referencing interfaces at https://github.com/typescript-eslint/typescript-eslint/blob/861844d727a2ba7f75cdbba2571498078aee2671/packages/typescript-estree/src/ts-estree/ast-node-types.ts and testing at https://runkit.com/brettz9/5d11892d0ed30f0013311bbe to confirm `TSInterfaceDeclaration` was indeed the type needed), I wasn't able to get it working there either.

```js
{
      code: `
          /**
           * foo.
           */
          interface quux {

          }
      `,
      errors: [
        {
          line: 3,
          message: 'JSDoc description does not satisfy the regex pattern.'
        }
      ],
      options: [
        {
          contexts: [
            'TSInterfaceBody',
            'TSInterfaceDeclaration',
            'TSInterfaceHeritage'
          ],
          noDefaults: true
        }
      ],
      parser: require.resolve('@typescript-eslint/parser')
    }
```

Likewise no success with `ClassProperty` testing against your `myClassField = 1`.

While I can understand it for typescript, since [`create`](https://eslint.org/docs/developer-guide/working-with-rules#rule-basics) mentions using "estree" and not "typescript-estree", I am surprised that `VariableDeclaration` and `Property` did not work to report a comment lacking above `myProp: true`. It appears ESlint only handles/passes on certain event types returned from `create`, though I'd need to investigate further.

**Comment by brettz9** [comment]
Nevermind--I see it should be possible, as we should just need to handle in our `getJSDocComment.js` file. I'll investigate further.

**Comment by brettz9** [comment]
PR #294 can now handle all types, including TypeScript's, with tests for all of the scenarios you mentioned.

The PR also adds such open-ended node type handling for `require-jsdoc` (to insist that there is even a jsdoc block at all) and for `require-description` (to insist that if a jsdoc block is present, that it has a description), noting that `match-description` only checks against a description when one is present.

**Comment by brettz9** [comment]
This has been fixed by #295 and is now available in [8.6.2](https://github.com/gajus/eslint-plugin-jsdoc/releases/tag/v8.6.2) (including on npm).

I had intended to have #294 fix this, leading me to neglect indicating this fact on #295 to tie the issues together, but this is indeed now fixed and can now be closed. (#294 will have a few more tests for this, but in the context of `match-description`.)
